NebulaWorld/SimulatedWorld.cs

```diff
@@ -395,6 +395,12 @@ public static void OnGameLoadCompleted()
                 LocalPlayer.Data.Mecha.TechBonuses.UpdateMech(GameMain.mainPlayer.mecha);
             }
 
+            //Initialization on the host side after game is loaded
+            if (LocalPlayer.IsMasterClient)
```

Comment:
This looks like a duplicated condition. The logic could simply exist in FactoryManager.

Reply:
I think in the `FactoryManager` it is not executed after the game fully loads. I think, when I place it in `Initialize()` method, it was executed before the `factory.prebuildPool` was initialized. So I needed to execute this after the game is fully loaded.

Reply:
Yeah, but the if statement is duplicated.better let FactoryManager do the masterclient check and not force that check into the callers.

Reply:
Fixed.